test/pipelinefinally_test.go

```diff
@@ -123,6 +180,26 @@ func TestPipelineLevelFinally_OneDAGTaskFailed_Failure(t *testing.T) {
 	if finalTaskStartTime.Before(dagTask1EndTime) || finalTaskStartTime.Before(dagTask2EndTime) {
 		t.Fatalf("Final Tasks should start getting executed after all DAG tasks finishes")
 	}
+
+	// two final tasks referring to results must be skipped
+	// finaltaskconsumingdagtask1 has a reference to a task result from failed task
+	// finaltaskconsumingdagtask4 has a reference to a task result from skipped task with when expression
+	expectedSkippedTasks := []v1beta1.SkippedTask{{
+		Name: "dagtask3",
+	}, {
+		Name:            "dagtask4",
+		WhenExpressions: we,
+	}, {
+		Name: "finaltaskconsumingdagtask1",
+	}, {
+		Name: "finaltaskconsumingdagtask4",
+	}}
+
+	if d := cmp.Diff(pr.Status.SkippedTasks, expectedSkippedTasks); d != "" {
+		t.Fatalf("Expected four skipped tasks, dag task with condition failure dagtask3, dag task with when expression,"+
+			"two final tasks with missing result reference finaltaskconsumingdagtask1 and finaltaskconsumingdagtask4 in SkippedTasks."+
+			" Diff: %s", diff.PrintWantGot(d))
+	}
 }
 
 func TestPipelineLevelFinally_OneFinalTaskFailed_Failure(t *testing.T) {
```

Comment:
do we need to update more than one end to end test? i feel like the previous one was pretty comprehensive, especially if it also has a finally task successfully execute while consuming a result

Reply:
nope, I think updating one is enough, I will add a finally task with a successful task result consumption in the first test itself.

Reply:
done

Reply:
verifying the param value in a finally task (referring to result of a dag task) https://github.com/tektoncd/pipeline/pull/3242/files#diff-7815ee756d1326fe39c09f4deb5c7c36aa4a8f14b4319d5bf8ac73a1929a3d10R210